python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED.py

```diff
@@ -0,0 +1,405 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::IAM::Role'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+
+def should_ignore_config_item(config_item, ignored_roles, ignored_users):
+    return (config_item['configurationItemStatus'] == 'ResourceDeleted')\
+            or (config_item['resourceType'] == 'AWS::IAM::Role' and config_item['resourceName'] in ignored_roles)\
+            or (config_item['resourceType'] == 'AWS::IAM::User' and config_item['resourceName'] in ignored_users)\
+            or (config_item['ARN'].rsplit("/")[1] == 'aws-service-role')
+
+
+def get_user_policies(username):

```

Comment:
Every API call is expensive for the service, please check iteratively if the policy is attach to the user, then if not the first group, then if not the second group etc. 
The goal is to reduce the number of call outside Config. Why not using the Configuration_item info, by the way?

Reply:
fair point, will switch to using the configuration item info where possible